### ISSUE TITLE
Move non-API related code into internal

### ISSUE THREAD

**Issue by lbernick** [tektoncd/pipeline#5679]
Based on this [discussion thread](https://github.com/tektoncd/pipeline/pull/5634#discussion_r997411776).
Everything in the [apis packages](https://github.com/tektoncd/pipeline/tree/main/pkg/apis) and the [client packages](https://github.com/tektoncd/pipeline/tree/main/pkg/client) should be public, and all other code should be moved into the internal folder where possible to avoid usage outside of the repo and reduce our Go API surface.

**Comment by vdemeester** [comment]
> Everything in the [apis packages](https://github.com/tektoncd/pipeline/tree/main/pkg/apis) and the [client packages](https://github.com/tektoncd/pipeline/tree/main/pkg/client) should be public, and all other code should be moved into the internal folder where possible to avoid usage outside of the repo and reduce our Go API surface.

Note that today, other tektoncd component might depend on those, for example the cli, or the operator, or triggers or even chains. I think for each package we want to move we need to see if it make sense or not to be "public" (and exported) or not. 

See https://github.com/tektoncd/pipeline/pull/5681#pullrequestreview-1154364716

**Comment by aThorp96** [comment]
I think this relates to https://github.com/tektoncd/pipeline/issues/10072 @vdemeester. LLM code "understanding" and vulnerability reports have almost always been, in my experience, lower quality due to internal APIs being public. Even outside of LLMs this will be very helpful, but LLMs I think raise the priority here if we want to curb false-positive vulnerability reports. 

We'll need to be conservative about which APIs we trust to be internal, since other Tekton components or third party extensions (custom resolvers, for example) may depend on some of the APIs as well. Still, chipping away at this will be helpful I think.

**Comment by vdemeester** [comment]
@aThorp96 agreed. I definitely think we need to make sure (and enforce) that by default we put any new package under `internal` if there is no valid reason to make it *public*.. Moving current public ones is a bit harder (as we may broke dependencies) so we need to be careful while doing it. But I 100% agree, especially in light of recent vulnerability reports (Q1/Q2 2026).
